airflow-core/tests/unit/plugins/test_plugin.py

```diff
@@ -44,45 +41,54 @@ def plugin_macro():
     pass
 
 
-# Creating a flask appbuilder BaseView
-class PluginTestAppBuilderBaseView(AppBuilderBaseView):
-    default_view = "test"
-
-    @expose("/")
-    def test(self):
-        return self.render_template("test_plugin/test.html", content="Hello galaxy!")
-
-
-v_appbuilder_view = PluginTestAppBuilderBaseView()
-v_appbuilder_package = {
-    "name": "Test View",
-    "category": "Test Plugin",
-    "view": v_appbuilder_view,
-    "label": "Test Label",
-}
-
-v_nomenu_appbuilder_package = {"view": v_appbuilder_view}
-
-# Creating flask appbuilder Menu Items
-appbuilder_mitem = {
-    "name": "Google",
-    "href": "https://www.google.com",
-    "category": "Search",
-}
-appbuilder_mitem_toplevel = {
-    "name": "apache",
-    "href": "https://www.apache.org/",
-    "label": "The Apache Software Foundation",
-}
+try:
+    from flask import Blueprint
+    from flask_appbuilder import BaseView as AppBuilderBaseView, expose
+    from starlette.middleware.base import BaseHTTPMiddleware
+
+    # Creating a flask appbuilder BaseView
+    class PluginTestAppBuilderBaseView(AppBuilderBaseView):
+        default_view = "test"
+
+        @expose("/")
+        def test(self):
+            return self.render_template("test_plugin/test.html", content="Hello galaxy!")
+
+    v_appbuilder_view = PluginTestAppBuilderBaseView()
+    v_appbuilder_package = {
+        "name": "Test View",
+        "category": "Test Plugin",
+        "view": v_appbuilder_view,
+        "label": "Test Label",
+    }
+
+    v_nomenu_appbuilder_package = {"view": v_appbuilder_view}
+
+    # Creating flask appbuilder Menu Items
+    appbuilder_mitem = {
+        "name": "Google",
+        "href": "https://www.google.com",
+        "category": "Search",
+    }
+    appbuilder_mitem_toplevel = {
+        "name": "apache",
+        "href": "https://www.apache.org/",
+        "label": "The Apache Software Foundation",
+    }
+
+    # Creating a flask blueprint to integrate the templates and static folder
+    bp = Blueprint(
+        "test_plugin",
+        __name__,
+        template_folder="templates",  # registers airflow/plugins/templates as a Jinja template folder
+        static_folder="static",
+        static_url_path="/static/test_plugin",
+    )
+
+except ImportError:
+    # This test plugin will be silently skipped if fab provider is not installed
+    pass
```

Comment:
I think it is uncool to silents skip. Can we make it a bit more explicit ynad only fail on explicit FAB things and not generally on all ImportErrors?